### PR TITLE
add timeAgo for reviews/verdicts

### PR DESCRIPTION
**Pull request by tdquan**

### CODE REVIEW

File: api/alembic/versions/ac8894d49876_add_science_feedback_published_date.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+                  sa.Column('scienceFeedbackPublishedDate',
+                            sa.DateTime()))
+
+
+def downgrade():
+    op.drop_column('appearance', 'scienceFeedbackPublishedDate')
+    op.drop_column('claim', 'scienceFeedbackPublishedDate')
+    op.drop_column('content', 'scienceFeedbackPublishedDate')
+    op.drop_column('review', 'scienceFeedbackPublishedDate')
+    op.drop_column('verdict', 'scienceFeedbackPublishedDate')
```
Comment (Ledoux) [inline api/alembic/versions/ac8894d49876_add_science_feedback_published_date.py:42]: :+1:

File: api/models/mixins/has_science_feedback_published_date.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+from sqlalchemy import Column, DateTime
+
+
+class HasScienceFeedbackPublishedDate:
+    scienceFeedbackPublishedDate = Column(DateTime())
```
Comment (Ledoux) [inline api/models/mixins/has_science_feedback_published_date.py:5]: lets put it simply in the HasScienceFeedbackMixin with the others science feedback properties, no need to create an other mixin :), what do you think ?

File: webapp/src/styles/components/layout/Feeds/Controls/_KeywordsBar.scss [hunk 3]
```diff
@@ -30,10 +29,20 @@
...
+    align-items: center;
     flex-direction: column;
+    width: 100%;
 
     .keywords-input {
-      margin-bottom: 1rem;
+      height: 2.75rem;
+      margin-bottom: 2rem;
+      width: 40rem;
+      max-width: 90%;
```
Comment (Ledoux) [inline webapp/src/styles/components/layout/Feeds/Controls/_KeywordsBar.scss:40]: alphabetical order please:)

File: api/domain/science_feedback/wordpress/claim_review.py [hunk 4]
```diff
@@ -45,11 +48,15 @@ def claim_review_from_url(url, session=None):
...
+
         return {
             'claim': {
                 'text': claimshort.text,
             },
             'conclusions': conclusions,
             'reviewers': reviewers,
-            'title': title.text
+            'title': title.text,
+            'publishedDate': published_date
```
Comment (Ledoux) [inline api/domain/science_feedback/wordpress/claim_review.py:61]: alphabetical order please

File: webapp/src/styles/components/layout/_VerdictItem.scss [hunk 5]
```diff
@@ -26,6 +30,7 @@
   }
 
   hr {
+    width: 25%;
```
Comment (Ledoux) [inline webapp/src/styles/components/layout/_VerdictItem.scss:33]: alphabetical order please

File: webapp/src/utils/timeAgo.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
+const _ = (date) => {
```
Comment (Ledoux) [inline webapp/src/utils/timeAgo.js:1]: Maybe we could use it more like a react hook to make it more memoizable ?

typically in components/uses/useTimeAgo.js

export default date => useMemo(() => <your function>, [date])

and in your AppearanceItem:
```
<strong>
     { useTimeAgo(publishedDate) }
</strong>
```


```

**Comment by Ledoux** [comment]
Very cool ! I put some suggestions :)

File: api/models/mixins/has_science_feedback_published_date.py (hunk 2 above)
Comment (tdquan) [reply api/models/mixins/has_science_feedback_published_date.py:5]: I tried that and the models that doesn't have this attribute crashed the app. May be there is a way around it that I don't know? @Ledoux

Comment (Ledoux) [reply api/models/mixins/has_science_feedback_published_date.py:5]: weird, there should be no special reason for that, let's put it back in the HasScienceFeedbackMixin and I will see locally what s the bug, ok ?

Comment (tdquan) [reply api/models/mixins/has_science_feedback_published_date.py:5]:
```
sqlalchemy.exc.ProgrammingError: (psycopg2.errors.UndefinedColumn) column medium.scienceFeedbackPublishedDate does not exist
```
something like this

Comment (Ledoux) [reply api/models/mixins/has_science_feedback_published_date.py:5]: but if you did not make the migration or reset your database with fb rmdb after, it is normal :).

Comment (tdquan) [reply api/models/mixins/has_science_feedback_published_date.py:5]: let me try again tmr. I think I tried this the first time when I just did the migration and it crashed so I had to move it to a separate file 😅

Comment (Ledoux) [reply api/models/mixins/has_science_feedback_published_date.py:5]: ok cool:),
don't forget to remove your with_models_creation=True the time to test the migration.
dans stamp fb alembic stamp <revisionId juste before this last one>
then fb alembic upgrade head

File: api/models/mixins/has_science_feedback_mixin.py [hunk 7]
```diff
@@ -1,7 +1,25 @@
...
     scienceFeedbackIdentifier = Column(String(32))
 
-    scienceFeedbackUrl = Column(String(512))
+    @declared_attr
+    def scienceFeedbackUrl(cls):
+        if cls.__tablename__ == 'verdict':
+            return Column(String(512))
+
+    @declared_attr
+    def scienceFeedbackPublishedDate(cls):
```
Comment (Ledoux) [inline api/models/mixins/has_science_feedback_mixin.py:23]: :+1:
